### DISCUSSION TITLE
Control the amount of logging in Directory Server

### DISCUSSION THREAD

**Discussion by corrados** [jamulussoftware/jamulus#3081] [Technical Support & Help]
Is there a possibility to control the amount of logging done at the Directory Server. I have very large log files on my server filled with entries in, e.g., daemon.log like this:
```
May 28 00:00:27 vs84947 Jamulus[319]: Requested to register entry for x.x.x.x:22124 (x.x.0.x:22124): Studio One (external)
May 28 00:00:41 vs84947 Jamulus[320]: Requested to register entry for x.x.x.x:22124 (x.x.0.2:22124): NJAMP Studio (external)
May 28 00:00:41 vs84947 Jamulus[321]: Requested to register entry for x.x.x.x:22124 (x.x.4.x:22124): Janne L Jam (external)
May 28 00:00:45 vs84947 Jamulus[320]: Requested to register entry for x.x.x.x:22124 (x.x.x.x:22124): Rob's Pi (external)
```
I would like to suppress this type of message in the logging if possible.

**Comment by pljones** [comment]
Do you need the log file enabled?

I don't use the log files for the directory logging, I just use the console output captured by systemd:
```
$ systemctl status Jamulus@Directory3
● Jamulus@Directory3.service - Directory on 22324
     Loaded: loaded (/etc/systemd/system/Jamulus@.service; enabled; vendor preset: enabled)
    Drop-In: /home/peter/Jamulus/systemd/Jamulus@Directory3.service.d
             └─Jamulus-Directory3.conf
     Active: active (running) since Thu 2023-06-01 18:41:35 BST; 1 week 0 days ago
       Docs: http://jamulus.drealm.info/jamulus
             https://jamulus.io/wiki/Server-Troubleshooting
   Main PID: 1088 (Jamulus-Directo)
      Tasks: 2 (limit: 18666)
     Memory: 38.2M
        CPU: 4min 55.235s
     CGroup: /system.slice/system-Jamulus.slice/Jamulus@Directory3.service
             └─1088 /opt/Jamulus/bin/Jamulus-Directory3 -n -s -p 22324 -u 1 -l /opt/Jamulus/log/Directory3.log -e localhost --serve>

Jun 09 18:29:49 fs-peter Jamulus-Directory3[1088]: Requested to register entry for 212.83.56.25:22124 (212.83.56.25:22124): 💎 Blue>
Jun 09 18:30:02 fs-peter Jamulus-Directory3[1088]: Requested to register entry for 92.206.190.96:43970 (192.168.1.166:22124): Mando>
Jun 09 18:30:18 fs-peter Jamulus-Directory3[1088]: Requested to register entry for 88.198.2.199:22124 (88.198.2.199:22124): My Serv>
Jun 09 18:30:34 fs-peter Jamulus-Directory3[1088]: Requested to register entry for 3.66.226.156:22124 (172.31.12.194:22124): MucMus>
Jun 09 18:30:51 fs-peter Jamulus-Directory3[1088]: Requested to register entry for 68.251.105.81:22124 (68.251.105.81:22124): tkap->
Jun 09 18:31:20 fs-peter Jamulus-Directory3[1088]: Requested to register entry for 51.210.8.130:22124 (51.210.8.130:22124): Standar>
Jun 09 18:31:31 fs-peter Jamulus-Directory3[1088]: Requested to register entry for 93.42.88.44:59406 (192.168.1.243:59406): 🌈Rainb>
Jun 09 18:32:30 fs-peter Jamulus-Directory3[1088]: Requested to register entry for 104.190.118.193:22123 (172.16.1.49:22123): 🍑Pea>
Jun 09 18:32:43 fs-peter Jamulus-Directory3[1088]: Requested to register entry for 75.6.179.251:22124 (127.0.0.1:22124): JazzJohnPi>
Jun 09 18:33:04 fs-peter Jamulus-Directory3[1088]: Requested to register entry for 62.113.206.158:22124 (62.113.206.158:22124): 🎸J>
```
That way the standard system log rotation, compression and cyclic deletion takes care of things.

**Reply by corrados** [reply]
I actually want to suppress all these console outputs because they spam my var/log :-). I could pipe all outputs to the Null device but I think it would be better to have control over the amount of console outputs with Jamulus command line arguments.

**Reply by rdica** [reply]
In your service file, maybe redirect the jamulus command output, ie;
`ExecStart=/path/to/jamulus $ARGS 2>&1 /dev/null`

If that doesn't work, put that redirected command in a script and call the script from `ExecStart`.

**Reply by corrados** [reply]
Yes, this is what I meant with "_I could pipe all outputs to the Null device_" in my comment above.

**Comment by ann0see** [comment]
I mean we could implement different log levels?
-l 0 --> debug logging (log everything)
-l 1 --> verbose logging/info logging
-l 2 --> only important stuff
etc.?

**Reply by corrados** [reply]
You are right. My request would only make sense if we have different log levels. If you would introduce just a flag like --omit_console_outputs, it would be the same as doing `[...] 2>&1 /dev/null`.

But after reading your suggestion, I was thinking what outputs I really want to have in the console output (i.e. in the system logs). These are outputs after a server crash or other severe issues. For all other logging I am using the already existing command line argument `-l`.

Since the Jamulus server is so reliable and never crashed so far, I think it is better not to change anything in the Jamulus code and I'll go for the `[...] 2>&1 /dev/null` solution on my server.

**Reply by pljones** [reply]
OK, so we're after some kind of log filter...
https://github.com/systemd/systemd/issues/6432 -> https://github.com/systemd/systemd/pull/24058
https://github.com/systemd/systemd/issues/6432#issuecomment-1382708218
Those are probably the relevant bits.  Looks like it's a relatively new `systemd` feature.
